RusimbiPatrick/Banka
# #165310829 built user registration endpoint

**What does this PR do?**
It creates a user registration endpoint.

**Description of Task to be done?**
1. It specifies the user registration route in the server.js file
2. It creates a route for user registration.
3. It creates a user registration controller for registering the user.
4. It creates a user registration service for registering the user.
5. It encrypts a submitted password during the registration.
6. It creates data structure arrays for storing user information.

**How should this be manually tested?**
Start the Server and POSTMAN.
Browse the following endpoint route: localhost:3000/api/v1/auth/signup.
Submit the required information with POST method.
Check if the validations work once you don't submit any of the required information.
After passing the validations, you should receive a 201 HTTP Status Code along with the information that you have submitted.
